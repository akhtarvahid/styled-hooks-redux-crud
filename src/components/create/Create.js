import React from "react";
import { Modal, Button, Form } from "react-bootstrap";
import './create.scss';
import { ADD_ITEM } from '../../utils/constant';

function Create({ status, handleClose,
    handleChange, submitCreateForm, state }) {
    const { title, description } = state;
    return (
        <Modal show={status} onHide={handleClose} animation={true} centered>
            <Modal.Header closeButton className='modal-content_add'>
                <Modal.Title>{ADD_ITEM}</Modal.Title>
            </Modal.Header>
            <Modal.Body>
                <Form onSubmit={submitCreateForm}>
                    <Form.Group controlId="formBasicTitle">
                        <Form.Label>Title</Form.Label>
                        <Form.Control
                            required
                            type="text"
                            placeholder="Enter Title"
                            name="title"
                            value={title || ''}
                            onChange={handleChange}
                        />
                    </Form.Group>
                    <Form.Group controlId="formBasicDescription">
                        <Form.Label>Description</Form.Label>
                        <Form.Control
                            required
                            type="text"
                            placeholder="Enter Description"
                            name="description"
                            value={description || ''}
                            onChange={handleChange}
                        />
                    </Form.Group>
                </Form>
            </Modal.Body>
            <Modal.Footer>
                <Button variant="danger" onClick={handleClose}>
                    Close
          </Button>
                <Button variant="success" onClick={submitCreateForm}>
                    Save
          </Button>
            </Modal.Footer>
        </Modal>
    );
}

export default Create;