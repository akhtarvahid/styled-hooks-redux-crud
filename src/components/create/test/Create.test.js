import React from 'react';
import renderer from 'react-test-renderer';
import Create from '../Create';

test('should render Create Component', () => {
    let fakeProps = {
        id: 1,
        title: "title",
        description: "description",
        completed: false
    };
    const tree = renderer
        .create(<Create state={fakeProps} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
