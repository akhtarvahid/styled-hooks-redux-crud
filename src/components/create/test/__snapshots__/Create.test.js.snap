// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Create Component 1`] = `null`;
